readme/BREAKING_CHANGES.markdown

```diff
@@ -1470,4 +1470,26 @@ This removal of the configuration has no replacement.
 
 ### Why was this change made?
 
-Liferay decided to not support this configuration.
\ No newline at end of file
+Liferay decided to not support this configuration.
+
+---------------------------------------
+
+## Removed extension points in SolrClientManager
+- **Date:** 2023-July-4
+- **JIRA Ticket:** [LPS-180691](https://liferay.atlassian.net/browse/LPS-180691)
+
+### What changed?
+
+These extension points in `SolrClientManager` for the `SolrClientFactory` with type `CLOUD` or `REPLICATED` and `HttpClientFactory` with type `BASIC` or `CERT` are being removed.
```

Comment:
@slnn I am confused, you are still having _basicAuthPoolingHttpClientFactory in your code, why you are saying `HttpClientFactory` with type `BASIC` is removed?

Reply:
@tinatian The extension points are being removed, meaning that developers can no longer deploy another service with the same type to override the Liferay provided ones.

Reply:
Hi @tinatian 

Yes, `SolrClientManager` now has a hard reference `_basicAuthPoolingHttpClientFactory`, however,  portal has a default implementation for the HttpClientFactory with type "BASIC", so it means that if anyone adds a new implementation for the HttpClientFactory with type "BASIC", we can not promise which service will be assigned to the reference `_basicAuthPoolingHttpClientFactory` which is with default ReferencePolicy.STATIC during startup since we can not control the start order between the default implementation and the new one.

```
@Reference(target = "(type=BASIC)")
	private HttpClientFactory _basicAuthPoolingHttpClientFactory;
```


By the way, even `SolrClientManager`  has this ServiceTrackerMap but it aims to track non-BASIC and non-CERT" HttpClientFactory, it also will not track the new implementation. 

```
_httpClientFactoryServiceTrackerMap =
			ServiceTrackerMapFactory.openSingleValueMap(
				bundleContext, HttpClientFactory.class,
				"(&(!(type=BASIC))(!(type=CERT)))",
				new PropertyServiceReferenceMapper<>("type"));
```

So I say these extension points for these SolrClientFactory and HttpClientFactory with special types are being removed.


Lily